torch/quantization/fx/utils.py

```diff
@@ -94,23 +94,30 @@ def get_per_tensor_qparams(activation_post_process):
     dtype = activation_post_process.dtype
     return scale, zero_point, dtype
 
-def get_quantize_op_and_qparams(activation_post_process):
+def get_quantize_node_info(activation_post_process: Callable) -> Tuple[str, Optional[Union[Callable, str]], Dict[str, Any]]:
     ''' Given an activation_post_process module,
-    return quantize op(e.g. quantize_per_tensor) and a dictionary
+    return node_type(e.g. call_function), quantize op(e.g. quantize_per_tensor) and a dictionary
     of extracted qparams from the module
     '''
-    scale, zero_point = activation_post_process.calculate_qparams()
     dtype = activation_post_process.dtype
-    if is_per_channel(activation_post_process.qscheme):
-        ch_axis = int(activation_post_process.ch_axis)
-        qparams = {'_scale_': scale, '_zero_point_': zero_point, '_axis_': ch_axis, '_dtype_': dtype}
-        quantize_op = torch.quantize_per_channel
-    else:
-        scale = float(scale)
-        zero_point = int(zero_point)
-        qparams = {'_scale_': scale, '_zero_point_': zero_point, '_dtype_': dtype}
-        quantize_op = torch.quantize_per_tensor  # type: ignore
-    return quantize_op, qparams
+    quantize_op : Optional[Union[Callable, str]] = None
+    if dtype in [torch.quint8, torch.qint8]:
+        node_type = "call_function"
+        scale, zero_point = activation_post_process.calculate_qparams()
+        if is_per_channel(activation_post_process.qscheme):
+            ch_axis = int(activation_post_process.ch_axis)
+            qparams = {"_scale_": scale, "_zero_point_": zero_point, "_axis_": ch_axis, "_dtype_": dtype}
+            quantize_op = torch.quantize_per_channel
+        else:
+            scale = float(scale)
+            zero_point = int(zero_point)
+            qparams = {"_scale_": scale, "_zero_point_": zero_point, "_dtype_": dtype}
+            quantize_op = torch.quantize_per_tensor  # type: ignore
+    elif dtype == torch.float16:
```

Comment:
this is for my own learning, some other places in this PR change the dtype on the observers from float16 to float32, where is this dtype==torch.float16 coming from?

Reply:
this is for supporting generating patterns for quantizing to fp16 precision (change of fp16 to fp32 is a fix that's unrelated to this check)